### PR TITLE
OWLS-69951 Add unit tests for Domain status processing and cleanup

### PR DESCRIPTION
**Pull request by russgold**
The rest of this task involves creating a wiki to recommend more status updates. The only thing that seems obvious, though, is semantic errors when processing the domain resource.

### CODE REVIEW

**Comment by russgold** [comment]
I've just found more code that should probably be tested as part of this PR, so let's not check it in yet.

**Comment by russgold** [comment]
In creating the new unit tests, I found a number of cases where the existing behavior appears to be wrong. I have marked them in the unit test with // todo tags

The replica status appears to make no sense. The tests `whenDomainUsesDefaultReplicaCount_statusLacksReplicaCount` and `whenDomainHasDefaultReplicaCount_statusReplicaCountIsSmallestNumberOfPodsInClustersWithoutReplica` describe what is currently happening. Perhaps it should set the replica count only when there is a single cluster? Or report for each cluster in the domain?

The logic to select servers to compute whether all servers are running seems to be backwards, selecting those in admin rather than running state.

When there is a pod failure, the code appears to want to delete any existing "available" conditions, but actually creates one if all servers are running.

My plan is to fix these once I have confirmation of the intended functionality and then to refactor the code in a subsequent commit, as cleaning it up would make it much easier to understand and modify.

**Comment by russgold** [comment]
Went ahead and did the cleanup, deciding to implement DomainStatus.replicas as undefined if there is not exactly one cluster in the domain, and otherwise the size of that cluster.

File: docs/domains/Domain.json [hunk 1]
```diff
@@ -297,6 +302,9 @@
           "type": "number",
           "minimum": 0
         },
+        "modified": {
```
Comment (rjeberhard) [inline docs/domains/Domain.json]: We shouldn't add this field.  The "modified" field in the processing logic was to just to track when the operator has a pending change to the status.

Comment (russgold) [reply docs/domains/Domain.json]: Now ignoring volatile fields in schema generation

**Pull request merged**